### PR TITLE
opcode SDIV and SMOD

### PR DESCRIPTION
**Pull request by icemelon**

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/execution/sdiv_smod.rs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        cb.add_constraint(
+            "overflow == 0 for opcode SDIV/SMOD",
+            mul_add_words.overflow(),
+        );
+
+        cb.condition(
+            (1.expr() - quotient_is_zero.expr()) * (1.expr() - divisor_is_zero.expr()),
+            |cb| {
+                cb.require_equal(
+                    "sgn(dividend) == sgn(residue) when quotient or divisor != 0 for opcode SDIV/SMOD",
```
Comment (silathdiir) [inline zkevm-circuits/src/evm_circuit/execution/sdiv_smod.rs:97]:
```suggestion
                    "sgn(dividend) == sgn(residue) when quotient or divisor == 0 for opcode SDIV/SMOD",
```

**Comment by silathdiir** [comment]
I will update this code after the spec code is merged.

**Review by icemelon — COMMENTED** [review]
@silathdiir You can rebase and create a draft PR to upstream. After the spec is merged, then mark the circuit pr ready to review.

**Comment by silathdiir** [comment]
Moved to upstream https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/559

**Pull request closed without merging**